### PR TITLE
[Security] Fix argon2 availability checks

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes (bug not released yet)
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

**Review by dunglas — APPROVED** [review]

File: src/Symfony/Component/Security/Core/Encoder/Argon2idPasswordEncoder.php [hunk 1]
```diff
@@ -82,4 +75,14 @@ public function isPasswordValid($encoded, $raw, $salt)
 
         throw new LogicException('Algorithm "argon2id" is not supported. Please install the libsodium extension or upgrade to PHP 7.3+.');
     }
+
+    /**
+     * @internal
+     */
+    public static function isDefaultSodiumAlgorithm()
```
Comment (teohhanhui) [inline src/Symfony/Component/Security/Core/Encoder/Argon2idPasswordEncoder.php]:
```suggestion
    public static function isDefaultSodiumAlgorithm(): bool
```

**Review by teohhanhui — CHANGES_REQUESTED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @chalasr.

**Comment by nicolas-grekas** [comment]
This PR makes no sense to me: we added a class for which we cannot guarantee that it will work depending on a default that is under control of libsodium only.
To me this is the sign that we should adopt the approach of libsodium instead: we should replace `Argon2idPasswordEncoder` by `LibsodiumPasswordEncoder` and align to its recommendation: trust them to always select the best default in the future. It's not like we have the choice: there is no other ways permitted by the extension (and I trust them on that it's the best).

Similarly, I would add a new `PhpPasswordEncoder` that would always use `PASSWORD_DEFAULT`, and deprecate `Argon2iPasswordEncoder` and `BCryptPasswordEncoder`.

**Comment by nicolas-grekas** [comment]
see #31016

**Pull request merged**